src/runtime_src/core/common/asd_parser.cpp

```diff
@@ -461,20 +475,26 @@ get_core_tile_info(aie_core_tile_status& core)
 
 bpt::ptree
 aie_core_tile_status::
-format_status(std::vector<aie_tiles_status>& aie_status, uint32_t start_col,
-              uint32_t cols, aie_tiles_info& tiles_info)
+format_status(const std::vector<aie_tiles_status>& aie_status, uint32_t cols, const aie_tiles_info& tiles_info,
+              uint32_t cols_filled)
 {
   bpt::ptree pt_array;
+  uint16_t col_count = 0;
```

Comment:
In the for loop below we use a different type from this one. I know we will never have uint16_t of aie cores (although that would be an interesting day if it does) these values should match.

There are a few places where this change should be made (the other for loops)
```suggestion
  uint32_t col_count = 0;
```